src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java

```diff
@@ -135,6 +146,56 @@
                         }));
     }
 
+    public Mono<ListResult<List<ConsentArtefactRepresentation>>> getAllConsentArtefacts(
+            String username, int limit, int offset) {
+        return Mono.create(monoSink -> dbClient.preparedQuery(SELECT_ALL_CONSENT_ARTEFACTS)
+                .execute(Tuple.of(username, limit, offset), handler -> {
+                    List<ConsentArtefactRepresentation> artefacts = getConsentArtefactRepresentation(monoSink, handler);
```

Comment:
Why pass the monoSink as argument to the method. Handle all success or error here, based on result.

Reply:
Agreed,  will make that change.